### PR TITLE
:seedling: [WIP] Rebase nits

### PR DESCRIPTION
**Pull request by mjudeikis**
* bump golang to 1.20
* replace all client imports for apiextension clients

### CODE REVIEW

File: pkg/server/scheme/scheme.go [hunk 1]
```diff
@@ -59,4 +61,80 @@ var (
...
+	GenericControlPlaneGroupName = "core"
+
+	// GenericControlPlaneSchemeGroupVersion is group version used to register these objects
+	GenericControlPlaneSchemeGroupVersion = schema.GroupVersion{Group: GenericControlPlaneGroupName, Version: "v1"}
 )
+
+// AddKnownTypes adds the set of types defined in the Kubernetes API to the Scheme.
+// This is minimal set of types that are required to be registered in the scheme in order to
+// serve KCP control plane usecases.
+func AddKnownTypes(scheme *runtime.Scheme) error {
```
Comment (mjudeikis) [inline pkg/server/scheme/scheme.go]: No idea if this wrong/right

File: go.mod [hunk 2]
```diff
@@ -174,6 +174,7 @@ require (
 )
 
 replace (
+	github.com/kcp-dev/client-go => ../client-go
```
Comment (mjudeikis) [inline go.mod:177]: This is wrong...

File: pkg/virtual/framework/rootapiserver/server.go [hunk 3]
```diff
@@ -80,7 +80,7 @@ func NewServer(c CompletedConfig, delegationTarget genericapiserver.DelegationTa
 
 func getRootHandlerChain(c CompletedConfig, delegateAPIServer genericapiserver.DelegationTarget) func(http.Handler, *genericapiserver.Config) http.Handler {
 	return func(apiHandler http.Handler, genericConfig *genericapiserver.Config) http.Handler {
-		delegateAfterDefaultHandlerChain := genericapiserver.DefaultBuildHandlerChain(
+		delegateAfterDefaultHandlerChain := genericapiserver.DefaultBuildHandlerChainBeforeAuthz( // Question:: should this be before or after authz?
```
Comment (sttts) [inline pkg/virtual/framework/rootapiserver/server.go]: double check that we don't lose any filter

File: pkg/server/options/flags.go [hunk 4]
```diff
@@ -138,26 +139,33 @@ var (
 		// features flags
 		"contention-profiling", // Enable lock contention profiling, if profiling is enabled
 		"profiling",            // Enable profiling via web interface host:port/debug/pprof/
+		"debug-socket-path",    //Use an unprotected (no authn/authz) unix-domain socket for profiling with the given path
```
Comment (sttts) [inline pkg/server/options/flags.go:142]: nit: space

File: pkg/server/options/flags.go [hunk 5]
```diff
@@ -138,26 +139,33 @@ var (
...
+		"peer-advertise-port",                   // If set and the UnknownVersionInteroperabilityProxy feature gate is enabled, this port will be used by peer kube-apiservers to proxy requests to this kube-apiserver when the request cannot be handled by the peer due to version skew between the kube-apiservers. This flag is only used in clusters configured with multiple kube-apiservers for high availability.
+		"peer-ca-file",                          // If set and the UnknownVersionInteroperabilityProxy feature gate is enabled, this file will be used to verify serving certificates of peer kube-apiservers. This flag is only used in clusters configured with multiple kube-apiservers for high availability.
+		"service-account-signing-key-file",      // Path to the file that contains the current private key of the service account token issuer. The issuer will sign issued ID tokens with this private key.
```
Comment (sttts) [inline pkg/server/options/flags.go:160]: odd position in misc

File: pkg/server/options/flags.go [hunk 6]
```diff
@@ -138,26 +139,33 @@ var (
...
+		"peer-advertise-ip",                     // If set and the UnknownVersionInteroperabilityProxy feature gate is enabled, this IP will be used by peer kube-apiservers to proxy requests to this kube-apiserver when the request cannot be handled by the peer due to version skew between the kube-apiservers. This flag is only used in clusters configured with multiple kube-apiservers for high availability.
+		"peer-advertise-port",                   // If set and the UnknownVersionInteroperabilityProxy feature gate is enabled, this port will be used by peer kube-apiservers to proxy requests to this kube-apiserver when the request cannot be handled by the peer due to version skew between the kube-apiservers. This flag is only used in clusters configured with multiple kube-apiservers for high availability.
+		"peer-ca-file",                          // If set and the UnknownVersionInteroperabilityProxy feature gate is enabled, this file will be used to verify serving certificates of peer kube-apiservers. This flag is only used in clusters configured with multiple kube-apiservers for high availability.
```
Comment (sttts) [inline pkg/server/options/flags.go:159]: I don't think we support these peer flags.

Also the aggregator routing flags should be hidden.

File: pkg/server/options/flags.go [hunk 7]
```diff
@@ -173,6 +181,7 @@ var (
 		"admission-control-config-file", // File with admission control configuration.
 		"disable-admission-plugins",     // admission plugins that should be disabled although they are in the default enabled plugins list (NamespaceLifecycle). Comma-delimited list of admission plugins: MutatingAdmissionWebhook, NamespaceLifecycle, ValidatingAdmissionWebhook. The order of plugins in this flag does not matter.
 		"enable-admission-plugins",      // admission plugins that should be enabled in addition to default enabled ones (NamespaceLifecycle). Comma-delimited list of admission plugins: MutatingAdmissionWebhook, NamespaceLifecycle, ValidatingAdmissionWebhook. The order of plugins in this flag does not matter.
+		"admission-control",             // Deprecated: Use --enable-admission-plugins or --disable-admission-plugins instead. Will be removed in a future version.
```
Comment (sttts) [inline pkg/server/options/flags.go:184]: we don't want to expose this.

File: pkg/informer/informer.go [hunk 8]
```diff
@@ -836,11 +836,13 @@ type crdGVRSource struct {
 var builtInInformableTypes map[schema.GroupVersionResource]GVRPartialMetadata = map[schema.GroupVersionResource]GVRPartialMetadata{
 	gvrFor("", "v1", "configmaps"):                                                          withGVRPartialMetadata(apiextensionsv1.NamespaceScoped, "ConfigMap", "configmap"),
 	gvrFor("", "v1", "events"):                                                              withGVRPartialMetadata(apiextensionsv1.NamespaceScoped, "Event", "event"),
+	gvrFor("", "v1", "limitranges"):                                                         withGVRPartialMetadata(apiextensionsv1.NamespaceScoped, "LimitRange", "limitranges"),
```
Comment (sttts) [inline pkg/informer/informer.go:839]: do we still expose limitranges? We shouldn't.

File: pkg/server/scheme/scheme.go (hunk 1 above)
Comment (sttts) [reply pkg/server/scheme/scheme.go]: don't like it. Why should we have our own set of types in a scheme? Why not just call the `Install` func for core?

Comment (mjudeikis) [reply pkg/server/scheme/scheme.go]: Me too. TBH was experimenting to make it running as it was not for me. 
Do we have `Install` function already?

File: pkg/virtual/framework/rootapiserver/server.go (hunk 3 above)
Comment (sttts) [reply pkg/virtual/framework/rootapiserver/server.go]: oh, this is a revert of https://github.com/sttts/kcp/pull/19/commits/0ab619e850b064057d68f1292601dbd56ccedee7 ?

Comment (sttts) [reply pkg/virtual/framework/rootapiserver/server.go]: can you squash those two?

**Pull request closed without merging**